File: treino/pilha e fila/PilhaCompleta.java
import java.util.Scanner;
import java.util.Stack;

class Celula {
    
    public int elemento;
    public Celula prox;

    public Celula() {
        this(0);
    }

    public Celula(int x) {
        this.elemento = x;
        this.prox = null;
    }
}

class Pilha {

    private Celula topo;

    public Pilha() {
        topo = null;
    }

   public void inserirTopo  ( int x) {
     Celula tmp = new Celula (x);
     topo.prox = tmp;
      topo= tmp ;
      tmp = null;
   }
   
    public int somar (){
   
   int total =0;
    Celula i;
   for( i = topo; i != null; i=i.prox ) {
   total =i.elemento ;

   }

 return total;
    }
   
   
   
    public void inserir(int x) {
        Celula tmp = new Celula(x); // crio uma celula temporaria para armazenar um elemento que vai ser inserido 
        tmp.prox = topo;   // faço a proxima celula temporaria apontar para topo 
        topo = tmp; // faço com que a celula topo aponte para a celula temporaria 
        tmp = null; // removo a celula temporaria 
    }

    public int remover() {
        if (topo != null) {
            int elemento = topo.elemento; // crio uma celula  para armazenar o elemento que  vai ser removido e esta no topo 
            Celula tmp = topo; // crio uma celula temporararia  para apontar para o elemento de  topo  
            topo = topo.prox; //aponto para o novo topo 
            tmp.prox = null;// desconecto o elemento da pilha 
            tmp = null; // limpo a celula temporaria 
            return elemento;
        }
        return -1;
    }

     public int soma (){
        int soma =0;
        for (Celula i =topo; i != null; i =i.prox){
            soma+=i.elemento;
                
        }
  return soma;

    }

    public void mostrar() {
        System.out.print("[");
        for (Celula i = topo; i != null; i = i.prox) {
            System.out.print(i.elemento + "");

        }
        System.out.println("]");

    }
}

 
public  static void ordem (Stack<integer> pilha ) { 
     if( pilha.topo == -1 ){
  return;
     }

 int elemento = pilha.pop();
  ordem(pilha);
  System.out.print(elemento + " ");
  pilha.push(elemento);

}







public class PilhaCompleta {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Pilha pilha = new Pilha();
        System.out.println(" digite o numero que vc quer inserir na pilha :      ( para sair digite -1)");
        int numero = sc.nextInt();
        while (numero != -1) {
            pilha.inserir(numero);
            numero = sc.nextInt();
        }
        System.out.println(" elementos da pilha : ");
        pilha.mostrar();
        System.out.println(" soma dos elementos da pilha : " + pilha.soma());
    }
}

File: treino/Bases/pilha e fila/PilhaCompleta.c
 /*
 * PilhaCompleta.c
 * Versão em C do PilhaCompleta.java
 * Implementa uma pilha ligada (lista encadeada) com operações básicas:
 * inserir (push), remover (pop), soma dos elementos e mostrar.
 */

#include <stdio.h>
#include <stdlib.h>

typedef struct Celula {
    int elemento;
    struct Celula* prox;
} Celula;

typedef struct Pilha {
    Celula* topo;
} Pilha;

Pilha* criarPilha(void) {
    Pilha* p = (Pilha*) malloc(sizeof(Pilha));
    if (!p) {
        fprintf(stderr, "Erro de alocacao\n");
        exit(EXIT_FAILURE);
    }
    p->topo = NULL;
    return p;
}

void inserir(Pilha* p, int x) {
    Celula* tmp = (Celula*) malloc(sizeof(Celula));
    if (!tmp) {
        fprintf(stderr, "Erro de alocacao\n");
        exit(EXIT_FAILURE);
    }
    tmp->elemento = x;
    tmp->prox = p->topo;
    p->topo = tmp;
}

int remover(Pilha* p) {
    if (p->topo == NULL) {
        return -1; // mesmo comportamento do Java
    }
    Celula* tmp = p->topo;
    int valor = tmp->elemento;
    p->topo = tmp->prox;
    free(tmp);
    return valor;
}

int soma(Pilha* p) {
    int s = 0;
    for (Celula* i = p->topo; i != NULL; i = i->prox) {
        s += i->elemento;
    }
    return s;
}

void mostrar(Pilha* p) {
    printf("[");
    for (Celula* i = p->topo; i != NULL; i = i->prox) {
        printf("%d", i->elemento);
        if (i->prox != NULL) printf(" ");
    }
    printf("]\n");
}

void liberarPilha(Pilha* p) {
    Celula* cur = p->topo;
    while (cur != NULL) {
        Celula* nxt = cur->prox;
        free(cur);
        cur = nxt;
    }
    free(p);
}

int main(void) {
    Pilha* pilha = criarPilha();
    int numero;

    printf("digite o numero que vc quer inserir na pilha : (para sair digite -1)\n");
    if (scanf("%d", &numero) != 1) {
        fprintf(stderr, "Entrada invalida\n");
        liberarPilha(pilha);
        return EXIT_FAILURE;
    }

    while (numero != -1) {
        inserir(pilha, numero);
        if (scanf("%d", &numero) != 1) break;
    }

    printf("elementos da pilha : \n");
    mostrar(pilha);
    printf("soma dos elementos da pilha : %d\n", soma(pilha));

    liberarPilha(pilha);
    return 0;
}
